articles/validating-xml-using-dtd/index.md

```diff
@@ -0,0 +1,250 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/validating-xml-using-dtd/
+title: Validating XML using DTD
+description: A tutorial on introduction to XML, the need for XML validation and how validation is done using DTD.
+author: srishilesh-p-s
+date: 2020-10-28T00:00:00-12:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/validating-xml-using-dtd/hero.jpg
+    alt: Validating XML using DTD
+---
+
+In this article, we will be learning about XML and how validation is being done using DTD. By the end of this article, you will understand different data serialization techniques, the need for XML, and the importance of XML validation. To understand the concepts better, you will also be validating a sample XML document.
+
+<!--more-->
+
+### Table of contents
+- [Introduction](#introduction)
+- [Overview of XML](#overview-of-xml)
+- [What is DTD?](#what-is-dtd)
+- [Need for XML validation](#need-for-xml-validation)
+- [Step by step guide for validation](#step-by-step-guide-for-validation)
+- [Conclusion](#conclusion)
+- [Further reading](#further-reading)
+
+### Introduction
+Computer systems vary in terms of hardware architecture, operating system, memory management mechanisms, and addressing architecture. Every data inside a computer is represented as binary values. Depending on the architecture, the representations vary. Similarly, the storage and communication mechanisms vary between each system. With each system having different architectures, communication between two different architectures requires a common medium for transportation of data. This leads us to the introduction to the concept of Data serialization
+
+![Data serialization](/engineering-education/validating-xml-using-dtd/data_serialization.jpg)
+
+[Image source](https://devopedia.org/data-serialization)
+
+> According to [Devopedia](https://devopedia.org/data-serialization), [Data serialization](https://en.wikipedia.org/wiki/Serialization) is the process of converting data objects present in complex data structures into byte streams for storage, transfer, and distribution purposes on physical devices.
+
+Generally, the data is stored as data structures like an array, tree, and heaps. For communication, the complex structures must be transformed into byte sequences. This series of bytes gets transported to the destination machine, to establish communication. The reverse process of Serialization is Deserialization.
```

Comment:
Since you have mentioned Deserialization here, it would be a good idea to give a formal definition with examples. It also would be a good idea to give some easily understandable examples.

Reply:
@cyberShaw I have added definition and a suitable example for understanding.